packages/dds/tree/src/test/shared-tree/undo.spec.ts

```diff
@@ -453,6 +464,137 @@ describe("Undo and redo", () => {
 		revertible.revert();
 		assert.equal(view.root.foo, 1);
 	});
+
+	it("reverts original & forked revertibles after making change to the original view", () => {
```

Comment:
Can you write these tests within the loop that all of the other tests (except 1 - "can undo while detached") in this file are in? If you look at the describe block at the top of the file, you will see that it runs all the tests in "attached" and "detached" modes. If you adapt your tests here to use the `getCheckout` helper, it should be easy to have them do the same thing.

Reply:
Well, actually, it's a little weird. You have a couple of options. 

1. Write your tests in the pattern already established in this file. So, you don't define a schema, and instead you do what all the other tests do and use `createCheckout` and `tree.editor` to do edits. This isn't the best way to edit the tree, but it's _consistent_ with how the rest of the file is doing it. So when somebody goes to convert this file later to the "better way", it won't really add any more work to change your tests since they'll all be updated in the same way.
2. Write your tests in a newer, better pattern - kind of what you are already doing, although I'd recommend you use something like `TestTreeProviderLite` rather than writing the factory/runtime code yourself. This is nice because it uses a real schema, and the better APIs for editing. However, then we end up with two different patterns in this file - the existing ones, and your new ones. And somebody coming along later to fix this file will have to deal with both patterns.

My vote would be for 1. We should either convert this whole file to a better test pattern first, and then do this PR. Or, we should write new tests in the same way as the current pattern, and then convert them all at once.

Curious what you think.

Reply:
I think I can go along with case (1) and I've noticed that there are skipped cases with `// TODO: unskip once forking revertibles is supported` comment. 

I can un-skip these pre-written test cases.

Reply:
@noencke 

I've tried to use `createCheckout()` and `tree.editor` but realized `clone: (branch?: TreeBranch)`, thus I can't (and shouldn't) pass `TreeCheckout`. Unless there is a way to convert `TreeCheckout` to `TreeBranch` prior to calling `clone()`, I think I should rewrite the skipped tests with my test design. 

What do you think?

Reply:
Just update `createCheckout()` to also return the SharedTree so that you can make a view from it via `viewWith()`. You can change `createCheckout()` however you want, really - since as we discussed above, we don't particularly like how any of this is currently operating.